flit_core/flit_core/tests/test_common.py

```diff
@@ -113,6 +113,8 @@ def test_normalize_file_permissions():
         ("<4, > 3.2", False),
         (">3.4", False),
         (">=2.7, !=3.0.*, !=3.1.*, !=3.2.*", True),
+        ("~3.3", False),
+        ("~2.7", True),
```

Comment:
`~3.3` is not actually a valid version specifier according to [PEP 440](https://peps.python.org/pep-0440/#version-specifiers), it should be `~=`.

I think we should make the different supported operators explicit, i.e. `(>=?|~=|==)`.